.github/workflows/codeql.yml

```diff
@@ -0,0 +1,41 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master", "dev" ]
+  pull_request:
+    branches: [ "master" ]
```

Comment:
I think we should let it run for any PR, not only the ones based in `master`.

Besides, we should also change the PR base to dev instead of master.